Lib/asyncio/proactor_events.py

```diff
@@ -426,6 +427,120 @@ def _pipe_closed(self, fut):
             self.close()
 
 
+class _ProactorDatagramTransport(_ProactorBasePipeTransport):
+
+    def __init__(self, loop, sock, protocol, address=None,
+                 waiter=None, extra=None):
+        super().__init__(loop, sock, protocol, waiter=waiter, extra=extra)
+        self._address = address
+        # We don't need to call _protocol.connection_made() since our base
+        # constructor does it for us.
+        self._buffer = collections.deque()
+        self._loop.call_soon(self._loop_reading)
+
+    def _set_extra(self, sock):
+        super()._set_extra(sock)
```

Comment:
I agree, I was initially confused by the name `_ProactorBasePipeTransport` and thought it was inaccurate. I think it would make sense to rename the class to `_ProactorBaseTransport`. I won't do it in this PR.